src/vcpkg/base/system.process.cpp

```diff
@@ -246,6 +251,118 @@ namespace vcpkg
 #endif
     }
 
+    Optional<ProcessStat> try_parse_process_stat_file(StringView text, StringView origin)
+    {
+        ParserBase p(text, origin);
+
+        p.match_while(ParserBase::is_ascii_digit); // pid %d (ignored)
+
+        p.skip_whitespace();
+        p.require_character('(');
+        // From: https://man7.org/linux/man-pages/man5/procfs.5.html
+        //
+        //  /proc/[pid]/stat
+        //
+        //  (2) comm  %s
+        //  The filename of the executable, in parentheses.
+        //  Strings longer than TASK_COMM_LEN (16) characters (including the terminating null byte) are silently
+        //  truncated.  This is visible whether or not the executable is swapped out.
+        const auto start = p.it().pointer_to_current();
+        const auto end = p.it().end();
+        size_t len = 0, last_seen = 0;
+        for (auto it = p.it(); len < 17 && it != end; ++len, ++it)
+        {
+            if (*it == ')') last_seen = len;
+        }
+        auto executable = StringView(start, last_seen);
+        for (size_t i = 0; i < last_seen; ++i)
+        {
+            p.next();
+        }
+        p.require_character(')');
+
+        p.skip_whitespace();
+        p.next(); // state %c (ignored)
+
+        p.skip_whitespace();
+        auto ppid_str = p.match_while(ParserBase::is_ascii_digit);
+        auto maybe_ppid = Strings::strto<int>(ppid_str);
+        if (auto ppid = maybe_ppid.get())
+        {
+            return ProcessStat{
+                *ppid,
+                executable.to_string(),
```

Comment:
```suggestion
        for (size_t i = 0; i < last_seen; ++i)
        {
            p.next();
        }
        p.require_character(')');

        p.skip_whitespace();
        p.next(); // state %c (ignored)

        p.skip_whitespace();
        auto ppid_str = p.match_while(ParserBase::is_ascii_digit);
        auto maybe_ppid = Strings::strto<int>(ppid_str);
        if (auto ppid = maybe_ppid.get())
        {
            return ProcessStat{
                *ppid,
                std::string(start, last_seen),
```

Extra StringView wasn't needed here.